src/features/governance/data/getProposals.ts

```diff
@@ -57,26 +71,56 @@ const runningProposalStages = new Set([ProposalStage.Queued, ProposalStage.Refer
 const pastProposalStages = new Set([ProposalStage.Expiration, ProposalStage.Execution]);
 
 export const getProposalRecord = async (
-  governance: GovernanceWrapper,
+  kit: ContractKit,
+  chainId: number,
   proposalID: string
 ): Promise<SerializedProposal | null> => {
-  const proposal = await governance.getProposalRecord(proposalID);
+  const multicall = getMultiCallForChain(
+    chainId,
+    kit.web3.eth.currentProvider as unknown as HttpProvider
+  );
+  const governance = await kit.contracts._web3Contracts.getGovernance();
+
+  const dequeueTx = governance.methods.getDequeue();
+  const stageTx = governance.methods.getProposalStage(proposalID);
+  const metadataTx = governance.methods.getProposal(proposalID);
+
+  const [dequeue, stage, metadata]: [string[], string, string[]] = await multicall.aggregate([
+    dequeueTx,
+    stageTx,
+    metadataTx,
+  ]);
+
   return {
-    ...jsonSafe(proposal),
     proposalID,
-    metadata: {
-      descriptionURL: proposal.metadata.descriptionURL,
-      timestamp: proposal.metadata.timestamp.toString(),
-    },
-  };
+    stage: PROPOSAL_STAGE_KEYS[parseInt(stage)] as ProposalStage,
+    metadata: { timestamp: metadata[2], descriptionURL: metadata[4] },
+    index: dequeue.findIndex((id) => proposalID === id),
+  } as SerializedProposal;
 };
 
-export async function getYamlForProposal(descriptionURL: string) {
-  const md = await fetch(getRawGithubUrl(descriptionURL))
-    .then((x) => x.text())
-    .catch(() => "Failed to fetch proposals' markdown");
-
-  return parsedYAMLFromMarkdown(md);
+export async function getYamlForProposal(proposal: MiniProposal) {
+  try {
+    const md = await fetch(getRawGithubUrl(proposal.metadata.descriptionURL)).then((x) => x.text());
+    const parsed = parsedYAMLFromMarkdown(md);
+    if (!parsed) {
+      throw new Error("Couldn't parse markdown");
+    }
+    return parsed;
+  } catch {
+    console.warn(`Failed to fetch proposal markdown at url ${proposal.metadata.descriptionURL}`);
+
+    return {
+      cgp: proposal.proposalID,
+      title: `Proposal ${proposal.proposalID}`,
+      dateCreated: proposal.metadata.timestamp,
+      author: '',
```

Comment:
if we arent getting author lets just leave it off the object

Reply:
We get it from the github yaml tho, this is just the fallback